framework/src/userobject/SolutionUserObject.C

```diff
@@ -288,8 +288,18 @@ SolutionUserObject::readExodusII()
   for (const auto & var_name : _elemental_variables)
     _system->add_variable(var_name, CONSTANT, MONOMIAL);
 
-  for (const auto & var_name : _scalar_variables)
-    _system->add_variable(var_name, FIRST, SCALAR);
+  // Handle scalars
+  if (_scalar_variables.size() > 0)
```

Comment:
Non-action comment: All the C++ containers have an `.empty()` method.

Reply:
I copied this straight from the existing code around line 374, but I didn't know about this method, will use this from now on